### PR TITLE
Made changes to eval cmd

### PR DESCRIPTION
**Pull request by tech6hutch**
### Description of the PR
Made changes to eval command, some discussed in chat.

### Changes Proposed in this Pull Request (List new items in CHANGELOG.MD)
- in `run`:
  - `type` can be either a Type object or undefined
  - remove error emitting block
- in `eval`:
  - destructure `flags` since it's used a lot
  - add error emitting to catch block

### Semver Classification

- [ ] This PR only includes documentation or non-code changes.
- [x] This PR fixes a bug and does not change the (intended) framework interface.
- [ ] This PR adds methods or properties to the framework interface.
- [ ] This PR removes or renames methods or properties in the framework interface.

### CODE REVIEW

File: src/commands/Admin/eval.js [hunk 1]
```diff
@@ -16,17 +16,11 @@ module.exports = class extends Command {
 
 	async run(msg, [code]) {
 		const { success, result, time, type } = await this.eval(msg, code);
-		const footer = this.client.methods.util.codeBlock('ts', type);
+		const footer = this.client.methods.util.codeBlock('ts', type ? type.toString() : '');
```
Comment (bdistin) [inline src/commands/Admin/eval.js]: remove the additional check here.

File: src/commands/Admin/eval.js [hunk 2]
```diff
@@ -63,15 +58,16 @@ module.exports = class extends Command {
 		} catch (error) {
 			if (!syncTime) syncTime = stopwatch.friendlyDuration;
```
Comment (bdistin) [inline src/commands/Admin/eval.js]: add `if (!type) type = new Type(error);` here

Comment (tech6hutch) [reply src/commands/Admin/eval.js]: Okay. That'll fix the type codeblock being empty when an error is thrown, too

File: src/commands/Admin/eval.js [hunk 3]
```diff
@@ -16,17 +16,11 @@ module.exports = class extends Command {
 
 	async run(msg, [code]) {
 		const { success, result, time, type } = await this.eval(msg, code);
-		const footer = this.client.methods.util.codeBlock('ts', type);
+		const footer = this.client.methods.util.codeBlock('ts', type.toString());
```
Comment (bdistin) [inline src/commands/Admin/eval.js]: unnecessary, util.codeBlock concatenates the object in a template literal, which automatically calls toString() on the object.

Comment (tech6hutch) [reply src/commands/Admin/eval.js]: Should the typings stay that it expects a string?

File: src/commands/Admin/eval.js [hunk 4]
```diff
@@ -16,18 +16,11 @@ module.exports = class extends Command {
 
 	async run(message, [code]) {
 		const { success, result, time, type } = await this.eval(message, code);
-		const footer = util.codeBlock('ts', type);
+		const footer = util.codeBlock('ts', type.toString());
```
Comment (bdistin) [inline src/commands/Admin/eval.js]: no toString() here

File: src/lib/util/util.js [hunk 5]
```diff
@@ -1,5 +1,6 @@
 const { promisify } = require('util');
 const { exec } = require('child_process');
+const DiscordUtil = require('discord.js').Util;
```
Comment (bdistin) [inline src/lib/util/util.js]: no

File: src/lib/util/util.js [hunk 6]
```diff
@@ -25,11 +26,15 @@ class Util {
...
 	 * @since 0.0.1
 	 * @param {string} lang The codeblock language
-	 * @param {string} expression The expression to be wrapped in the codeblock
+	 * @param {StringResolvable} expression The expression to be wrapped in the codeblock
 	 * @returns {string}
 	 */
 	static codeBlock(lang, expression) {
-		return `\`\`\`${lang}\n${expression || zws}\`\`\``;
+		// The resolved string form of expression, but never empty
+		const resolved = expression ?
```
Comment (bdistin) [inline src/lib/util/util.js]: no

File: src/lib/util/util.js [hunk 7]
```diff
@@ -25,11 +26,15 @@ class Util {
...
+			DiscordUtil.resolveString(expression) || zws :
```
Comment (bdistin) [inline src/lib/util/util.js]: no

File: src/lib/util/util.js [hunk 8]
```diff
@@ -25,11 +26,15 @@ class Util {
...
+			zws;
```
Comment (bdistin) [inline src/lib/util/util.js]: no

File: src/lib/util/util.js [hunk 9]
```diff
@@ -25,11 +26,15 @@ class Util {
...
+		return `\`\`\`${lang}\n${resolved}\`\`\``;
```
Comment (bdistin) [inline src/lib/util/util.js]: revert

**Pull request merged**